src/main/java/org/folio/service/orders/flows/update/unopen/UnOpenCompositeOrderManager.java

```diff
@@ -113,29 +127,49 @@ private Future<Void> processInventory(CompositePoLine compPOL, RequestContext re
           logger.debug(String.format("Pieces were removed : %s", deletedIds));
         }
       })
-        .mapEmpty();
+      .mapEmpty();
     } else if (PoLineCommonUtil.isItemsUpdateRequired(compPOL)) {
-      return processInventoryWithItems(compPOL, requestContext);
+        if (compPOL.getCheckinItems()) {
+          return (deleteHoldings) ? processInventoryOnlyWithHolding(compPOL, requestContext) : Future.succeededFuture();
+        } else {
+          return (deleteHoldings) ? processInventoryWithItems(compPOL, requestContext) : processInventoryOnlyWithItems(compPOL, requestContext);
+        }
     } else if (PoLineCommonUtil.isHoldingsUpdateRequired(compPOL)) {
```

Comment:
We should use similar check for holdings as well, createItem can be as INSTANCE_HOLDING and in this case we can have connected holdings, I assume we should write our check as following:
if (PoLineCommonUtil.isHoldingsUpdateRequired(compPOL)) {
 if (compPOL.getCheckinItems()) { // independent
    (deleteHoldings) ? processInventoryOnlyWithHolding(compPOL, requestContext) : Future.succeededFuture();
} else { // synchronized 
  processInventoryOnlyWithHolding(compPOL, requestContext);
}

Reply:
added code